src/Umbraco.Web.Website/Controllers/UmbRegisterController.cs

```diff
@@ -36,59 +51,109 @@ public async Task<IActionResult> HandleRegisterMember([Bind(Prefix = "registerMo
                 return CurrentUmbracoPage();
             }
 
+            MergeRouteValuesToModel(model);
+
             // U4-10762 Server error with "Register Member" snippet (Cannot save member with empty name)
             // If name field is empty, add the email address instead.
             if (string.IsNullOrEmpty(model.Name) && string.IsNullOrEmpty(model.Email) == false)
             {
                 model.Name = model.Email;
             }
```

Comment:
This could be part of `RegisterMemberAsync` as well, in case the method will be reused in the future